addons/website_sale/controllers/product_configurator.py

```diff
@@ -1,147 +1,263 @@
+            )
+            basic_product_information.update({
+                'price': price,
+                'can_be_sold': not (request.website.prevent_zero_price_sale and has_zero_price),
+                # The following fields are needed for tracking.
+                'category_name': product_or_template.categ_id.name,
+                'currency_name': currency.name,
+            })
+            if strikethrough_price:
+                basic_product_information['strikethrough_price'] = strikethrough_price
+        return basic_product_information
+
+    def _get_ptav_price_extra(self, ptav, currency, so_date, product_or_template):
+        price_extra = super()._get_ptav_price_extra(ptav, currency, so_date, product_or_template)
+        if request.is_frontend:
+            return self._apply_taxes_to_price(price_extra, product_or_template, currency)
+        return price_extra
+
+    def _get_strikethrough_price(self, product_or_template, pricelist, currency, date, price):
+        # Don't apply taxes to `compare_list_price`, it should be displayed as is.
+        compare_list_price = product_or_template.currency_id._convert(
+            from_amount=product_or_template.compare_list_price,
+            to_currency=currency,
+            company=request.env.company,
+            date=date,
+            round=False,
+        )
+        list_price = self._apply_taxes_to_price(
+            product_or_template._price_compute('list_price', currency=currency)[
+                product_or_template.id
+            ],
+            product_or_template,
+            currency
+        )
+
+        # First, try to use `compare_list_price` as the strikethrough price.
+        if (
+            request.env.user.has_group('website_sale.group_product_price_comparison')
+            and compare_list_price
+        ):
+            # If `compare_list_price` is lower than `price`, don't show it, but don't fall back on
+            # `list_price` either.
+            if currency.compare_amounts(compare_list_price, price) == 1:
+                return compare_list_price
+        # Second, try to use `list_price` as the strikethrough price.
+        elif (
+            pricelist.discount_policy == 'without_discount'
+            and currency.compare_amounts(list_price, price) == 1
+        ):
+            return list_price
+
+        return None
+
+    def _show_product(self, product_template, parent_combination):
+        if request.is_frontend:
+            return product_template._is_add_to_cart_possible(parent_combination)
+        return super()._show_product(product_template, parent_combination)
+
+    @staticmethod
+    def _populate_currency_and_pricelist(kwargs):
+        if request.is_frontend:
+            website = request.website
+            kwargs.update({
+                'currency_id': website.currency_id.id,
+                'pricelist_id': website.pricelist_id.id,
+            })
```

Comment:
I don't believe `request.is_frontend` is needed here. I expect this method to be called from the website anyway. If you believe it's still required, please put the check outside the method for clarity.
```suggestion
          website = request.website
          kwargs.update({
              'currency_id': website.currency_id.id,
              'pricelist_id': website.pricelist_id.id,
```

Reply:
Indeed, this method will only ever be called for frontend requests, so we can remove the check. FYI, my intention was to make it clear that we only "override" the currency and pricelist for requests coming from the frontend (but I'm fine with removing it).